### ISSUE TITLE
error occur when start SASServer1_1 and 2_1

### ISSUE THREAD

**Issue by zhanglianbo35** [sassoftware/sas-wsm#1]
some errors will occur
![image](https://user-images.githubusercontent.com/19187724/180972814-c9f06f54-2255-4ba1-bcf3-0da532f400c7.png)

**Comment by ajforeman** [comment]
Just set up 1.0.0 release on a new system and was unable to reproduce, but I've got an idea on what's happening.

The array built as laststop should be full of lines from a WebAppServer's server.log file that show "_Stopping Service [Catalina]_", of which we then keep the last value in the array, so that we know the line number in the log when the WebAppServer was last stopped.

You will notice in the code for this (copied below) that we are referencing the location of the server.log file, which is partially built from other variables. Of these, the user is expected to have set the SAS Configuration Directory path as **$sasconfigpath** variable:

```
#pull a list of line numbers from the WebAppServer's server log, matching string indicating server stop operation, then keep only the last member of the list (the latest matching line number)
$laststop = Select-String -Path "$sasconfigpath\Web\WebAppServer\$webappsvrname\logs\server.log" -Pattern 'Stopping service \[Catalina\]' | select-object -ExpandProperty LineNumber
$laststop = $laststop[-1]
```

Please check to see if that **$sasconfigpath** and the path to the WebAppServer log file is defined correctly. It may be a case of there not being a server.log file found at the location it's looking for, so when the array index -1 is referenced, there is nothing in the array (because no matches were read from the non-existent file) and thus the NullArray error you saw happens. In a future release I'll add something to catch this and not try assigning on an empty array.

Should that not be the fix please reply back and let me know.

**Comment by ajforeman** [comment]
Ok, I actually ran into this today. It looks like it can happen if your server.log file rolls, resulting in there never being a `Stopping Service [Catalina]` message in the "new" server.log file.

In my case it did successfully start the service even though PowerShell threw the NullArray errors. So while it doesn't _look_ great, it shouldn't cause any functional problems.

The fix for this will be the same as I said yesterday... just need to catch if there's no results from the search and handle accordingly (so we don't try assigning on nothing). Will update and push when I have some time.

**Comment by ajforeman** [comment]
This should be fixed as of SAS-wsm 1.0.1 release.

Closing issue.
